presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeAdjacentWindows.java

```diff
@@ -46,7 +47,16 @@
     @Override
     public Optional<PlanNode> apply(WindowNode parent, Captures captures, Context context)
     {
-        WindowNode child = captures.get(CHILD);
+        // Pulling the descendant WindowNode above projects is done as a part of this rule, as opposed in a
+        // separate rule, because that pullup is not useful on its own, and could be undone by other rules.
+        // For example, a rule could insert a project-off node between adjacent WindowNodes that use different
+        // input symbols.
```

Comment:
Imagine we have a rule that does the pull-up operation that the helper function does.  Then:
WA is a WindowNode that adds A = f(B).
WC is a WindowNode that adds C = f(D), with the same window specification as WA.
We start with WA over WC.
1.  Unused column pruning will introduce a ProjectNode P between them to project off D.
2.  A standalone WindowNode pull-up rule moves WC above P, adding in an identity projection for D.
3.  Any of three things are now possible:
* Elimination of P as it is now a no-op identity projection.
* Merging of WA and WC (what we wanted).
* Reinsertion of another ProjectNode between WA and WC, projecting off D again (which would block the rewrite that we wanted).